### PR TITLE
🌐 [i18n-KO] Translated `main_classes/model.md` to Korean

### PR DESCRIPTION
**Pull request by fabxoe**
# What does this PR do?

Translated the `main_classes/model.md` file of the documentation to Korean.
Thank you in advance for your review. 

Part of https://github.com/huggingface/transformers/issues/20179

## Before reviewing
- [x] Check for missing / redundant translations (번역 누락/중복 검사)
- [x] Grammar Check (맞춤법 검사)
- [x] Review or Add new terms to glossary (용어 확인 및 추가)
- [x] Check Inline TOC (e.g. `[[lowercased-header]]`)
- [ ] Check live-preview for gotchas (live-preview로 정상작동 확인)

## Who can review? (Initial)

@chhaewxn, @ahnjj, @jun048098, @fabxoe, @nuatmochoi, @heuristicwave

## Before submitting
- [x] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [x] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review? (Final)

@stevhliu May you please review this PR?

### CODE REVIEW

File: docs/source/ko/main_classes/model.md [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+specific language governing permissions and limitations under the License.
+
+⚠️ Note that this file is in Markdown but contain specific syntax for our doc-builder (similar to MDX) that may not be
+rendered properly in your Markdown viewer.
+
+-->
+
+# 모델
+
+기본 클래스 [`PreTrainedModel`], [`TFPreTrainedModel`], [`FlaxPreTrainedModel`]는 로컬 파일과 디렉토리로부터 모델을 로드하고 저장하거나 또는 (허깅페이스 AWS S3 리포지토리로부터 다운로드된) 라이브러리에서 제공하는 사전훈련 모델 설정값을 로드하고 저장하는 것을 지원하는 기본 메소드를 구현하였습니다.
```
Comment (chhaewxn) [inline docs/source/ko/main_classes/model.md:19]:
```suggestion
기본 클래스 [`PreTrainedModel`], [`TFPreTrainedModel`], [`FlaxPreTrainedModel`]는 로컬 파일과 디렉토리로부터 모델을 로드하고 저장하거나 또는 (허깅페이스 AWS S3 리포지토리로부터 다운로드된) 라이브러리에서 제공하는 사전 훈련된 모델 설정을 로드하고 저장하는 것을 지원하는 기본 메소드를 구현하였습니다.   
```

File: docs/source/ko/main_classes/model.md [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+-->
+
+# 모델
+
+기본 클래스 [`PreTrainedModel`], [`TFPreTrainedModel`], [`FlaxPreTrainedModel`]는 로컬 파일과 디렉토리로부터 모델을 로드하고 저장하거나 또는 (허깅페이스 AWS S3 리포지토리로부터 다운로드된) 라이브러리에서 제공하는 사전훈련 모델 설정값을 로드하고 저장하는 것을 지원하는 기본 메소드를 구현하였습니다.   
+
+[`PreTrainedModel`]과 [`TFPreTrainedModel`]은 또한 모든 모델들에 공통적으로 지원하는 몇가지 메소드를 구현하였습니다:
+
+- 새 토큰이 단어장에 추가될 때, 입력 토큰 임베딩의 크기를 조정합니다.
+- 모델의 어텐션 헤드를 가지치기합니다.
```
Comment (chhaewxn) [inline docs/source/ko/main_classes/model.md:24]:
```suggestion
- 모델의 어텐션 헤드를 가지치기 합니다.
```

File: docs/source/ko/main_classes/model.md [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+- [`~generation.GenerationMixin`](파이토치 모델용)
+- [`~generation.FlaxGenerationMixin`](Flax/JAX 모델용)
+
+## PreTrainedModel
+
+[[autodoc]] PreTrainedModel
+    - push_to_hub
+    - all
+
+사용자정의 모델은 superfast init이 특정모델에 적용될 수 있는지 여부를 결정하는 `_supports_assign_param_buffer`도 포함해야합니다.
```
Comment (chhaewxn) [inline docs/source/ko/main_classes/model.md:38]:
```suggestion
사용자 정의 모델은 초고속 초기화가 특정 모델에 적용될 수 있는지 여부를 결정하는 `_supports_assign_param_buffer`도 포함해야 합니다.
```
superfast init도 번역을 하면 좋을 것 같아 제안드립니다!😊

**Review by chhaewxn — COMMENTED** [review]
양질의 번역을 다수 해주셔서 많이 배우고 갑니다🤗👍

File: docs/source/ko/main_classes/model.md (hunk 1 above)
Comment (ahnjj) [inline docs/source/ko/main_classes/model.md:19]:
```suggestion
기본 클래스 [`PreTrainedModel`], [`TFPreTrainedModel`], [`FlaxPreTrainedModel`]는 로컬 파일 및 디렉토리에서 모델을 로드하고 저장하거나,
 또는 허깅페이스 AWS S3 리포지토리에서 다운로드한 라이브러리에서 제공하는 사전훈련 모델 설정값으로부터 모델을 로드하고 저장하는 기본 메소드를 구현하였습니다.   
```
가독성을 높이려 표현을 바꿔보았습니다 !

File: docs/source/ko/main_classes/model.md [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+⚠️ Note that this file is in Markdown but contain specific syntax for our doc-builder (similar to MDX) that may not be
+rendered properly in your Markdown viewer.
+
+-->
+
+# 모델
+
+기본 클래스 [`PreTrainedModel`], [`TFPreTrainedModel`], [`FlaxPreTrainedModel`]는 로컬 파일과 디렉토리로부터 모델을 로드하고 저장하거나 또는 (허깅페이스 AWS S3 리포지토리로부터 다운로드된) 라이브러리에서 제공하는 사전훈련 모델 설정값을 로드하고 저장하는 것을 지원하는 기본 메소드를 구현하였습니다.   
+
+[`PreTrainedModel`]과 [`TFPreTrainedModel`]은 또한 모든 모델들에 공통적으로 지원하는 몇가지 메소드를 구현하였습니다:
```
Comment (ahnjj) [inline docs/source/ko/main_classes/model.md:21]:
```suggestion
[`PreTrainedModel`]과 [`TFPreTrainedModel`]은 모든 모델들에 공통적으로 지원되는 몇 메소드를 구현하였습니다:
```

File: docs/source/ko/main_classes/model.md [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+- [`~generation.FlaxGenerationMixin`](Flax/JAX 모델용)
+
+## PreTrainedModel
+
+[[autodoc]] PreTrainedModel
+    - push_to_hub
+    - all
+
+사용자정의 모델은 superfast init이 특정모델에 적용될 수 있는지 여부를 결정하는 `_supports_assign_param_buffer`도 포함해야합니다.
+`test_save_and_load_from_pretrained`이 실패하면, 당신의 모델이 `_supports_assign_param_buffer`를 필요로 하는지 확인하세요.
```
Comment (ahnjj) [inline docs/source/ko/main_classes/model.md:39]:
```suggestion
`test_save_and_load_from_pretrained` 실패 시, 모델이 `_supports_assign_param_buffer`를 필요로 하는지 확인하세요.
```

**Review by ahnjj — CHANGES_REQUESTED** [review]

File: docs/source/ko/main_classes/model.md [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+# 모델
+
+기본 클래스 [`PreTrainedModel`], [`TFPreTrainedModel`], [`FlaxPreTrainedModel`]는 로컬 파일과 디렉토리로부터 모델을 로드하고 저장하거나 또는 (허깅페이스 AWS S3 리포지토리로부터 다운로드된) 라이브러리에서 제공하는 사전훈련 모델 설정값을 로드하고 저장하는 것을 지원하는 기본 메소드를 구현하였습니다.   
+
+[`PreTrainedModel`]과 [`TFPreTrainedModel`]은 또한 모든 모델들에 공통적으로 지원하는 몇가지 메소드를 구현하였습니다:
+
+- 새 토큰이 단어장에 추가될 때, 입력 토큰 임베딩의 크기를 조정합니다.
+- 모델의 어텐션 헤드를 가지치기합니다.
+
+각 모델에 공통인 다른 메소드들은 다음과 같은 클래스에 정의 되어있습니다.
```
[the remark this answers is not in the document]
Comment (jun048098) [reply docs/source/ko/main_classes/model.md:26]: 표현을 바꾸었습니다.

**Review by jun048098 — COMMENTED** [review]
fix: docs suggestion

File: docs/source/ko/main_classes/model.md [hunk 7]
```diff
@@ -0,0 +1,68 @@
...
+- [`~generation.GenerationMixin`](파이토치 모델용)
+- [`~generation.FlaxGenerationMixin`](Flax/JAX 모델용)
+
+## PreTrainedModel
+
+[[autodoc]] PreTrainedModel
+    - push_to_hub
+    - all
+
+사용자 정의 모델은 초고속 초기화가 특정 모델에 적용될 수 있는지 여부를 결정하는 `_supports_assign_param_buffer`도 포함해야 합니다.
```
Comment (fabxoe) [inline docs/source/ko/main_classes/model.md:38]:
```suggestion
사용자 정의 모델은 초고속 초기화(superfast init)가 특정 모델에 적용될 수 있는지 여부를 결정하는 `_supports_assign_param_buffer`도 포함해야 합니다.
```

File: docs/source/ko/main_classes/model.md [hunk 8]
```diff
@@ -0,0 +1,68 @@
...
+⚠️ Note that this file is in Markdown but contain specific syntax for our doc-builder (similar to MDX) that may not be
+rendered properly in your Markdown viewer.
+
+-->
+
+# 모델
+
+기본 클래스 [`PreTrainedModel`], [`TFPreTrainedModel`], [`FlaxPreTrainedModel`]는 로컬 파일과 디렉토리로부터 모델을 로드하고 저장하거나 또는 (허깅페이스 AWS S3 리포지토리로부터 다운로드된) 라이브러리에서 제공하는 사전 훈련된 모델 설정을 로드하고 저장하는 것을 지원하는 기본 메소드를 구현하였습니다.   
+
+[`PreTrainedModel`]과 [`TFPreTrainedModel`]은 또한 모든 모델들에 공통적으로 지원하는 몇가지 메소드를 구현하였습니다:
```
Comment (fabxoe) [inline docs/source/ko/main_classes/model.md:21]:
```suggestion
[`PreTrainedModel`]과 [`TFPreTrainedModel`]은 또한 모든 모델들을 공통적으로 지원하는 메소드 여러개를 구현하였습니다:
```

**Review by stevhliu — APPROVED** [review]
Thanks, we can merge once the conflict is resolved! 🤗

**Pull request merged**